diff --git a/Gulpfile.js b/Gulpfile.js
index c8a60079b..b49041ae6 100644
--- a/Gulpfile.js
+++ b/Gulpfile.js
@@ -4,7 +4,7 @@ const plumber = require("gulp-plumber");
 const through = require("through2");
 const chalk = require("chalk");
 const newer = require("gulp-newer");
-const babel = require("gulp-babel");
+const sucrase = require("@sucrase/gulp-plugin");
 const gulpWatch = require("gulp-watch");
 const fancyLog = require("fancy-log");
 const filter = require("gulp-filter");
@@ -66,7 +66,7 @@ function buildBabel(exclude, sourcesGlob = defaultSourcesGlob) {
     .pipe(errorsLogger())
     .pipe(newer({ dest: base, map: swapSrcWithLib }))
     .pipe(compilationLogger())
-    .pipe(babel())
+    .pipe(sucrase({transforms: ["flow", "imports"]}))
     .pipe(
       // Passing 'file.relative' because newer() above uses a relative
       // path and this keeps it consistent.
@@ -105,7 +105,7 @@ function buildRollup(packages) {
   );
 }
 
-const bundles = ["packages/babel-parser"];
+const bundles = [];
 
 gulp.task("build-rollup", () => buildRollup(bundles));
 gulp.task("build-babel", () => buildBabel(/* exclude */ bundles));
diff --git a/Makefile b/Makefile
index 2c077a3fb..dfa6ff5e7 100644
--- a/Makefile
+++ b/Makefile
@@ -129,7 +129,9 @@ test-only:
 	BABEL_ENV=test ./scripts/test.sh
 	$(MAKE) test-clean
 
-test: lint test-only
+test:
+	make bootstrap
+	make test-only
 
 test-ci: jest-ci
 
@@ -207,7 +209,14 @@ endif
 	rm -f .npmrc
 	$(MAKE) clean
 
-bootstrap-only: lerna-bootstrap
+bootstrap-only: clean-all
+	yarn --ignore-engines
+	yarn link sucrase
+	yarn link @sucrase/gulp-plugin
+	yarn link @sucrase/jest-plugin
+	yarn link @sucrase/webpack-loader
+	yarn link @sucrase/webpack-object-rest-spread-plugin
+	./node_modules/.bin/lerna bootstrap
 
 yarn-install: clean-all
 	yarn --ignore-engines
diff --git a/package.json b/package.json
index a420748f1..7678c4796 100644
--- a/package.json
+++ b/package.json
@@ -91,6 +91,9 @@
     }
   },
   "jest": {
+    "transform": {
+      ".js": "@sucrase/jest-plugin"
+    },
     "collectCoverageFrom": [
       "packages/*/src/**/*.mjs",
       "packages/*/src/**/*.js",
diff --git a/packages/babel-core/src/transformation/index.js b/packages/babel-core/src/transformation/index.js
index e2eccd894..e0d501153 100644
--- a/packages/babel-core/src/transformation/index.js
+++ b/packages/babel-core/src/transformation/index.js
@@ -59,7 +59,7 @@ export function runSync(
   try {
     transformFile(file, config.passes);
   } catch (e) {
-    e.message = `${opts.filename ?? "unknown"}: ${e.message}`;
+    e.message = `${opts.filename != null ? opts.filename : "unknown"}: ${e.message}`;
     if (!e.code) {
       e.code = "BABEL_TRANSFORM_ERROR";
     }
@@ -72,7 +72,7 @@ export function runSync(
       ({ outputCode, outputMap } = generateCode(config.passes, file));
     }
   } catch (e) {
-    e.message = `${opts.filename ?? "unknown"}: ${e.message}`;
+    e.message = `${opts.filename != null ? opts.filename : "unknown"}: ${e.message}`;
     if (!e.code) {
       e.code = "BABEL_GENERATE_ERROR";
     }
diff --git a/packages/babel-core/test/config-chain.js b/packages/babel-core/test/config-chain.js
index 078cd91dc..a01ad07fc 100644
--- a/packages/babel-core/test/config-chain.js
+++ b/packages/babel-core/test/config-chain.js
@@ -7,7 +7,7 @@ import { loadOptions as loadOptionsOrig } from "../lib";
 // TODO: In Babel 8, we can directly uses fs.promises which is supported by
 // node 8+
 const pfs =
-  fs.promises ??
+  fs.promises != null ? fs.promises :
   new Proxy(fs, {
     get(target, name) {
       if (name === "copyFile") {
diff --git a/packages/babel-helper-builder-react-jsx/src/index.js b/packages/babel-helper-builder-react-jsx/src/index.js
index 64b136923..52d948427 100644
--- a/packages/babel-helper-builder-react-jsx/src/index.js
+++ b/packages/babel-helper-builder-react-jsx/src/index.js
@@ -95,7 +95,9 @@ You can turn on the 'throwIfNamespace' flag to bypass this warning.`,
       value.value = value.value.replace(/\n\s+/g, " ");
 
       // "raw" JSXText should not be used from a StringLiteral because it needs to be escaped.
-      delete value.extra?.raw;
+      if (value.extra != null) {
+        delete value.extra.raw;
+      }
     }
 
     if (t.isJSXNamespacedName(node.name)) {
diff --git a/packages/babel-helper-create-regexp-features-plugin/src/index.js b/packages/babel-helper-create-regexp-features-plugin/src/index.js
index 49554be42..412d6c909 100644
--- a/packages/babel-helper-create-regexp-features-plugin/src/index.js
+++ b/packages/babel-helper-create-regexp-features-plugin/src/index.js
@@ -24,7 +24,7 @@ export function createRegExpFeaturePlugin({ name, feature, options = {} }) {
     name,
     pre() {
       const { file } = this;
-      const features = file.get(featuresKey) ?? 0;
+      const features = file.get(featuresKey) != null ? file.get(featuresKey) : 0;
       let newFeatures = enableFeature(features, FEATURES[feature]);
 
       const { useUnicodeFlag, runtime = true } = options;
@@ -49,7 +49,7 @@ export function createRegExpFeaturePlugin({ name, feature, options = {} }) {
         const { node } = path;
         const { file } = this;
         const features = file.get(featuresKey);
-        const runtime = file.get(runtimeKey) ?? true;
+        const runtime = file.get(runtimeKey) != null ? file.get(runtimeKey) : true;
         const regexpuOptions = generateRegexpuOptions(node, features);
         if (regexpuOptions === null) {
           return;
diff --git a/packages/babel-helper-module-transforms/src/rewrite-live-references.js b/packages/babel-helper-module-transforms/src/rewrite-live-references.js
index 8bd93f509..7514f4f5f 100644
--- a/packages/babel-helper-module-transforms/src/rewrite-live-references.js
+++ b/packages/babel-helper-module-transforms/src/rewrite-live-references.js
@@ -240,7 +240,7 @@ const rewriteReferencesVisitor = {
 
         const exportedNames = exported.get(localName);
         const importData = imported.get(localName);
-        if (exportedNames?.length > 0 || importData) {
+        if ((exportedNames != null ? exportedNames.length : undefined) > 0 || importData) {
           assert(path.node.operator === "=", "Path was not simplified");
 
           const assignment = path.node;
diff --git a/packages/babel-parser/src/parser/expression.js b/packages/babel-parser/src/parser/expression.js
index 89088db4d..a9e94a28c 100644
--- a/packages/babel-parser/src/parser/expression.js
+++ b/packages/babel-parser/src/parser/expression.js
@@ -876,7 +876,7 @@ export default class ExpressionParser extends LValParser {
       node,
       call.arguments,
       true,
-      call.extra?.trailingComma,
+      call.extra != null ? call.extra.trailingComma : undefined,
     );
     return node;
   }
diff --git a/packages/babel-parser/src/parser/lval.js b/packages/babel-parser/src/parser/lval.js
index 547441b0c..66292c434 100644
--- a/packages/babel-parser/src/parser/lval.js
+++ b/packages/babel-parser/src/parser/lval.js
@@ -70,7 +70,7 @@ export default class LValParser extends NodeUtils {
             if (
               isLast &&
               prop.type === "RestElement" &&
-              node.extra?.trailingComma
+              node.extra != null ? node.extra.trailingComma : undefined
             ) {
               this.raiseRestNotLast(node.extra.trailingComma);
             }
@@ -96,7 +96,7 @@ export default class LValParser extends NodeUtils {
             node.elements,
             isBinding,
             contextDescription,
-            node.extra?.trailingComma,
+            node.extra != null ? node.extra.trailingComma : undefined,
           );
           break;
 
diff --git a/packages/babel-parser/src/plugins/flow.js b/packages/babel-parser/src/plugins/flow.js
index ab5f646b6..aed12c839 100644
--- a/packages/babel-parser/src/plugins/flow.js
+++ b/packages/babel-parser/src/plugins/flow.js
@@ -953,7 +953,7 @@ export default (superClass: Class<Parser>): Class<Parser> =>
             variance,
             kind,
             allowSpread,
-            allowInexact ?? !exact,
+            allowInexact != null ? allowInexact : !exact,
           );
 
           if (propOrInexact === null) {
@@ -1870,7 +1870,7 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         ((node.params: any): N.Expression[]),
         true,
         "arrow function parameters",
-        node.extra?.trailingComma,
+        node.extra != null ? node.extra.trailingComma : undefined,
       );
       // Enter scope, as checkParams defines bindings
       this.scope.enter(functionFlags(false, false) | SCOPE_ARROW);
diff --git a/packages/babel-parser/src/util/scopeflags.js b/packages/babel-parser/src/util/scopeflags.js
index d92699883..2bdedecaa 100644
--- a/packages/babel-parser/src/util/scopeflags.js
+++ b/packages/babel-parser/src/util/scopeflags.js
@@ -2,18 +2,20 @@
 
 // Each scope gets a bitset that may contain these flags
 // prettier-ignore
-export const SCOPE_OTHER        = 0b0000000000,
-             SCOPE_PROGRAM      = 0b0000000001,
-             SCOPE_FUNCTION     = 0b0000000010,
-             SCOPE_ASYNC        = 0b0000000100,
-             SCOPE_GENERATOR    = 0b0000001000,
-             SCOPE_ARROW        = 0b0000010000,
-             SCOPE_SIMPLE_CATCH = 0b0000100000,
-             SCOPE_SUPER        = 0b0001000000,
-             SCOPE_DIRECT_SUPER = 0b0010000000,
-             SCOPE_CLASS        = 0b0100000000,
-             SCOPE_TS_MODULE    = 0b1000000000,
-             SCOPE_VAR = SCOPE_PROGRAM | SCOPE_FUNCTION | SCOPE_TS_MODULE;
+export const SCOPE_OTHER        = 0b0000000000;
+// TODO(#455): This was changed to simpler code to work with Sucrase. Fix the
+// Sucrase bug to avoid this patch hunk.
+export const SCOPE_PROGRAM      = 0b0000000001;
+export const SCOPE_FUNCTION     = 0b0000000010;
+export const SCOPE_ASYNC        = 0b0000000100;
+export const SCOPE_GENERATOR    = 0b0000001000;
+export const SCOPE_ARROW        = 0b0000010000;
+export const SCOPE_SIMPLE_CATCH = 0b0000100000;
+export const SCOPE_SUPER        = 0b0001000000;
+export const SCOPE_DIRECT_SUPER = 0b0010000000;
+export const SCOPE_CLASS        = 0b0100000000;
+export const SCOPE_TS_MODULE    = 0b1000000000;
+export const SCOPE_VAR = SCOPE_PROGRAM | SCOPE_FUNCTION | SCOPE_TS_MODULE;
 
 export type ScopeFlags =
   | typeof SCOPE_OTHER
@@ -38,39 +40,39 @@ export function functionFlags(isAsync: boolean, isGenerator: boolean) {
 
 // These flags are meant to be _only_ used inside the Scope class (or subclasses).
 // prettier-ignore
-export const BIND_KIND_VALUE           = 0b00000_0000_01,
-             BIND_KIND_TYPE            = 0b00000_0000_10,
+export const BIND_KIND_VALUE           = 0b00000_0000_01;
+export const BIND_KIND_TYPE            = 0b00000_0000_10;
              // Used in checkLVal and declareName to determine the type of a binding
-             BIND_SCOPE_VAR            = 0b00000_0001_00, // Var-style binding
-             BIND_SCOPE_LEXICAL        = 0b00000_0010_00, // Let- or const-style binding
-             BIND_SCOPE_FUNCTION       = 0b00000_0100_00, // Function declaration
-             BIND_SCOPE_OUTSIDE        = 0b00000_1000_00, // Special case for function names as
+export const BIND_SCOPE_VAR            = 0b00000_0001_00; // Var-style binding
+export const BIND_SCOPE_LEXICAL        = 0b00000_0010_00; // Let- or const-style binding
+export const BIND_SCOPE_FUNCTION       = 0b00000_0100_00; // Function declaration
+export const BIND_SCOPE_OUTSIDE        = 0b00000_1000_00; // Special case for function names as
                                                    // bound inside the function
              // Misc flags
-             BIND_FLAGS_NONE           = 0b00001_0000_00,
-             BIND_FLAGS_CLASS          = 0b00010_0000_00,
-             BIND_FLAGS_TS_ENUM        = 0b00100_0000_00,
-             BIND_FLAGS_TS_CONST_ENUM  = 0b01000_0000_00,
-             BIND_FLAGS_TS_EXPORT_ONLY = 0b10000_0000_00;
+export const BIND_FLAGS_NONE           = 0b00001_0000_00;
+export const BIND_FLAGS_CLASS          = 0b00010_0000_00;
+export const BIND_FLAGS_TS_ENUM        = 0b00100_0000_00;
+export const BIND_FLAGS_TS_CONST_ENUM  = 0b01000_0000_00;
+export const BIND_FLAGS_TS_EXPORT_ONLY = 0b10000_0000_00;
 
 // These flags are meant to be _only_ used by Scope consumers
 // prettier-ignore
 /*                              =    is value?    |    is type?    |      scope          |    misc flags    */
-export const BIND_CLASS         = BIND_KIND_VALUE | BIND_KIND_TYPE | BIND_SCOPE_LEXICAL  | BIND_FLAGS_CLASS  ,
-             BIND_LEXICAL       = BIND_KIND_VALUE | 0              | BIND_SCOPE_LEXICAL  | 0                 ,
-             BIND_VAR           = BIND_KIND_VALUE | 0              | BIND_SCOPE_VAR      | 0                 ,
-             BIND_FUNCTION      = BIND_KIND_VALUE | 0              | BIND_SCOPE_FUNCTION | 0                 ,
-             BIND_TS_INTERFACE  = 0               | BIND_KIND_TYPE | 0                   | BIND_FLAGS_CLASS  ,
-             BIND_TS_TYPE       = 0               | BIND_KIND_TYPE | 0                   | 0                 ,
-             BIND_TS_ENUM       = BIND_KIND_VALUE | BIND_KIND_TYPE | BIND_SCOPE_LEXICAL  | BIND_FLAGS_TS_ENUM,
-             BIND_TS_AMBIENT    = 0               | 0              | 0            | BIND_FLAGS_TS_EXPORT_ONLY,
+export const BIND_CLASS         = BIND_KIND_VALUE | BIND_KIND_TYPE | BIND_SCOPE_LEXICAL  | BIND_FLAGS_CLASS  ;
+export const BIND_LEXICAL       = BIND_KIND_VALUE | 0              | BIND_SCOPE_LEXICAL  | 0                 ;
+export const BIND_VAR           = BIND_KIND_VALUE | 0              | BIND_SCOPE_VAR      | 0                 ;
+export const BIND_FUNCTION      = BIND_KIND_VALUE | 0              | BIND_SCOPE_FUNCTION | 0                 ;
+export const BIND_TS_INTERFACE  = 0               | BIND_KIND_TYPE | 0                   | BIND_FLAGS_CLASS  ;
+export const BIND_TS_TYPE       = 0               | BIND_KIND_TYPE | 0                   | 0                 ;
+export const BIND_TS_ENUM       = BIND_KIND_VALUE | BIND_KIND_TYPE | BIND_SCOPE_LEXICAL  | BIND_FLAGS_TS_ENUM;
+export const BIND_TS_AMBIENT    = 0               | 0              | 0            | BIND_FLAGS_TS_EXPORT_ONLY;
              // These bindings don't introduce anything in the scope. They are used for assignments and
              // function expressions IDs.
-             BIND_NONE          = 0               | 0              | 0                   | BIND_FLAGS_NONE   ,
-             BIND_OUTSIDE       = BIND_KIND_VALUE | 0              | 0                   | BIND_FLAGS_NONE   ,
+export const BIND_NONE          = 0               | 0              | 0                   | BIND_FLAGS_NONE   ;
+export const BIND_OUTSIDE       = BIND_KIND_VALUE | 0              | 0                   | BIND_FLAGS_NONE   ;
 
-             BIND_TS_CONST_ENUM = BIND_TS_ENUM | BIND_FLAGS_TS_CONST_ENUM,
-             BIND_TS_NAMESPACE  = 0               | 0              | 0            | BIND_FLAGS_TS_EXPORT_ONLY;
+export const BIND_TS_CONST_ENUM = BIND_TS_ENUM | BIND_FLAGS_TS_CONST_ENUM;
+export const BIND_TS_NAMESPACE  = 0               | 0              | 0            | BIND_FLAGS_TS_EXPORT_ONLY;
 
 export type BindingTypes =
   | typeof BIND_NONE
diff --git a/packages/babel-polyfill/src/index.js b/packages/babel-polyfill/src/index.js
index aa1b4639f..cca92db5a 100644
--- a/packages/babel-polyfill/src/index.js
+++ b/packages/babel-polyfill/src/index.js
@@ -7,7 +7,7 @@ if (global._babelPolyfill && typeof console !== "undefined" && console.warn) {
     "@babel/polyfill is loaded more than once on this page. This is probably not desirable/intended " +
       "and may have consequences if different versions of the polyfills are applied sequentially. " +
       "If you do need to load the polyfill more than once, use @babel/polyfill/noConflict " +
-      "instead to bypass the warning.",
+      "instead to bypass the warning."
   );
 }
 
diff --git a/packages/babel-register/src/cache.js b/packages/babel-register/src/cache.js
index b505bb494..804bdb19b 100644
--- a/packages/babel-register/src/cache.js
+++ b/packages/babel-register/src/cache.js
@@ -17,7 +17,7 @@ let data: Object = {};
 let cacheDisabled = false;
 
 function isCacheDisabled() {
-  return process.env.BABEL_DISABLE_CACHE ?? cacheDisabled;
+  return process.env.BABEL_DISABLE_CACHE != null ? process.env.BABEL_DISABLE_CACHE : cacheDisabled;
 }
 /**
  * Write stringified cache to disk.
diff --git a/scripts/gulp-tasks.js b/scripts/gulp-tasks.js
index 73c1f7db2..59c35bfcf 100644
--- a/scripts/gulp-tasks.js
+++ b/scripts/gulp-tasks.js
@@ -23,6 +23,8 @@ const DuplicatePackageCheckerPlugin = require("duplicate-package-checker-webpack
 const WarningsToErrorsPlugin = require("warnings-to-errors-webpack-plugin");
 const webpackStream = require("webpack-stream");
 const uglify = require("gulp-uglify");
+const ObjectRestSpreadPlugin = require("@sucrase/webpack-object-rest-spread-plugin");
+
 
 function webpackBuild(opts) {
   const plugins = opts.plugins || [];
@@ -41,12 +43,8 @@ function webpackBuild(opts) {
       rules: [
         {
           test: /\.js$/,
-          loader: "babel-loader",
-          options: {
-            // Use the bundled config so that module syntax is passed through
-            // for Webpack.
-            envName: "standalone",
-          },
+          loader: "@sucrase/webpack-loader",
+          options: {transforms: ["imports", "flow"]},
         },
       ],
     },
@@ -63,6 +61,7 @@ function webpackBuild(opts) {
       libraryTarget: "umd",
     },
     plugins: [
+      new ObjectRestSpreadPlugin(),
       new WarningsToErrorsPlugin(),
       new DuplicatePackageCheckerPlugin({
         exclude(instance) {
